src/main/java/de/tum/in/www1/artemis/web/rest/ogparser/LinkPreviewResource.java

```diff
@@ -0,0 +1,40 @@
+package de.tum.in.www1.artemis.web.rest.ogparser;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.cache.annotation.Cacheable;
+import org.springframework.web.bind.annotation.*;
+
+import de.tum.in.www1.artemis.security.annotations.EnforceAtLeastStudent;
+import de.tum.in.www1.artemis.service.linkpreview.LinkPreviewService;
+import de.tum.in.www1.artemis.web.rest.dto.LinkPreviewDTO;
+
+/**
+ * REST controller for Link Preview.
+ */
+@RestController
+@RequestMapping("/api")
+public class LinkPreviewResource {
+
+    private final Logger log = LoggerFactory.getLogger(LinkPreviewResource.class);
+
+    private final LinkPreviewService linkPreviewService;
+
+    public LinkPreviewResource(LinkPreviewService linkPreviewService) {
+        this.linkPreviewService = linkPreviewService;
+    }
+
+    /**
+     * GET /link-preview : get link preview for url.
+     *
+     * @param url the url to parse
+     * @return the LinkPreviewDTO containing the meta information
+     */
+    @PostMapping("/link-preview")
```

Comment:
This can be a GET, right?

Reply:
Actually i think no, this endpoint cannot be a GET request because it expects the URL to be passed in the request body. In RESTful API design, GET requests typically do not have request bodies. 

This can be get request only if i pass the url as param.